tools/ClassGen/NodeBuilder.cpp

```diff
@@ -293,12 +335,23 @@ void NodeBuilder::emitEdges(std::ostream &os) const {
   }
   os << "  llvm_unreachable(\"Invalid index\");\n}\n";
 
-  os << "\nllvm::StringRef " << name_
+  os << "\nstd::string " << name_
      << "Node::getOutputName(unsigned idx) const {\n";
+  // Handle fixed outputs.
   for (size_t i = 0; i < nodeOutputs_.size(); i++) {
     os << "  if (idx == " << i << ") { return \"" << nodeOutputs_[i].second
        << "\"; }\n";
   }
+  // Handle variable outputs.
+  os << "  idx -= " << nodeOutputs_.size() << ";\n";
```

Comment:
Is this done to use nodeOutput-relative indexing?

Reply:
Yes, It indeed makes _idx_ relative to the current variable output. From this, it's simpler to detect of the index corresponds to a value of this particular output and to generate a name for the value (that contains the relative index).